client/cli/src/params/pruning_params.rs

```diff
@@ -23,21 +23,30 @@ use sc_service::{BlocksPruning, PruningMode};
 /// Parameters to define the pruning mode
 #[derive(Debug, Clone, PartialEq, Args)]
 pub struct PruningParams {
-	/// Specify the state pruning mode, a number of blocks to keep or 'archive'.
+	/// Specify the state pruning mode.
 	///
-	/// Default is to keep only the last 256 blocks,
-	/// otherwise, the state can be kept for all of the blocks (i.e 'archive'),
-	/// or for all of the canonical blocks (i.e 'archive-canonical').
+	/// This mode specifies when the block's state (ie, storage)
+	/// should be pruned (ie, removed) from the database.
+	///
+	/// Options available:
+	///   'archive'            Keep the state of all blocks.
+	///   'archive-canonical'  Keep only the state of finalized blocks.
+	///   number               Keep the state of the last number of finalized blocks.
+	///
+	/// The default option is to keep the last 256 blocks (number == 256).
 	#[arg(alias = "pruning", long, value_name = "PRUNING_MODE")]
 	pub state_pruning: Option<String>,
```

Comment:
We should change this to:
```suggestion
	#[arg(alias = "pruning", long, value_name = "PRUNING_MODE", default = "256")]
	pub state_pruning: PruningMode,
```

Then add the following enum:

```
enum PruningMode {
     Archive,
     ArchiveCanonical,
     Custom(u32),
}

impl clap::ValueEnum for PruningMode {
     .. impl the enum
}
```

Reply:
Yep, that does make sense! Thanks for the suggestion!

One problem with `ValueEnum` is that its happy with unit only enum variants. This is related to the [value_variants](https://docs.rs/clap/latest/clap/trait.ValueEnum.html#tymethod.value_variants) that requires us to state the expected values. If the `Custom(32)` is omitted, the default value of "256" cannot be provided anymore.

To overcome this I've changed the `ValueEnum` to `FromStr`.
Let me know what you think of it